### ISSUE TITLE
Linting error

### ISSUE THREAD

**Issue by alastairbattrick** [hashicorp/vscode-terraform#188]
Hi
Version: 1.3.12

The linter indicates an error, despite terraform not having an error when parsing the code.

```
resource "aws_route" "route-vpn-private" {
  count                  = "${length(var.office_subnets) * length(aws_route_table.private)}"
  destination_cidr_block = "${element(element(setproduct(var.office_subnets, aws_route_table.private), count.index),0)}"
  route_table_id         = "${element(element(setproduct(var.office_subnets, aws_route_table.private), count.index),1).id}"
  gateway_id             = "${aws_vpn_gateway.vpn_gateway.id}"
}
```
The route_table_id line is highlighted with
```
expected "}" but found "."
```

I reported this to the linter repo, but they don't think it is a problem with them:
https://github.com/wata727/tflint/issues/330

Thanks

**Comment by mauve** [comment]
Can you post the full error message please?

**Comment by alastairbattrick** [comment]
Hi
Does this help?
![image](https://user-images.githubusercontent.com/5707483/59612332-cf9c2100-9114-11e9-97d2-e245cb79104c.png)

**Comment by jpreese** [comment]
We are getting the same thing

![image](https://user-images.githubusercontent.com/6980197/59983761-8e62b000-95f1-11e9-9608-980ce71a83ce.png)

**Comment by phil-bevan** [comment]
@jpreese  I see the same as you - are you using Terraform 0.12?  The linter cannot handle 'resource[index].property'syntax which is required in TF 0.12.  If you switch to the TF 0.11 syntax of element(resource.*.property,index) then the linter will not error (but TF 0.12 will)

**Comment by jpreese** [comment]
Yes, this is on 0.12 . I've been updating to 0.12 syntax where possible and just ignoring the linting errors

**Comment by manishrathi4** [comment]
I have the same issue. What is the fix other than ignoring? Can't go back to older version of TF. Will this be fixed in new version of plugin?

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
